deepinv/tests/test_optim.py

```diff
@@ -196,6 +196,58 @@ def A_adjoint(v):
         assert torch.allclose(xhat, x)
 
 
+def test_l12_prior():
+    device = dinv.utils.get_freer_gpu() if torch.cuda.is_available() else "cpu"
+
+    # create sparse ground truth
+    x = torch.zeros(30, 30)
+    x = torch.fill(x, 0.05)  # probability for bernoulli
+    n_chan = 3  # length of l2 axis
+    x = torch.bernoulli(x)
+    x = x.unsqueeze(0).unsqueeze(0).to(device)
+    x = x.tile(1, n_chan, 1, 1)
+    x[:, 0, ::] *= 0.7
+    x[:, 1, ::] *= 0.5
+    x[:, 2, ::] *= 0.2
+
+    physics = dinv.physics.Blur(
+        dinv.physics.blur.gaussian_blur(sigma=(0.7, 0.7), angle=0.0),
+        padding="circular",
+        device=device,
+        noise_model=dinv.physics.noise.GaussianNoise(sigma=0.01),
+    )
+
+    y = physics(x)
+    max_iter = 1000
```

Comment:
Hi @Nils-Laurent, did you find a way to reduce the number of iterations in the test?

Reply:
Hi @tachella, I created this test because it requires a low number of iteration to converge. I noticed that L12 in test_priors_algo triggered the assert because the L2 axis was wrong, when I set it to the channels' axis, it passes the test

Reply:
Maybe I should simply use L12Prior in test_priors_algo, that would make the PR more concise